frame/contracts/src/gas.rs

```diff
@@ -181,17 +185,44 @@ impl<T: Config> GasMeter<T> {
 		self.gas_left = self.gas_left.saturating_add(adjustment).min(self.gas_limit);
 	}
 
+	/// This method is used for gas syncs with the engine.
+	///
+	/// Updates internal `engine_comsumed` tracker of engine fuel consumption.
+	///
+	/// Charges self with the `ref_time` Weight corresponding to wasmi fuel consumed on the engine
+	/// side since last sync. Passed value is scaled by multiplying it by the weight of a basic
+	/// operation, as such an operation in wasmi engine costs 1.
+	///
+	/// Returns the updated `gas_left` `Weight` value from the meter.
+	/// Normally this would never fail, as engine should fail first when out of gas.
+	pub fn sync_fuel(&mut self, wasmi_fuel: u64) -> Result<Weight, DispatchError> {
+		let wasmi_fuel = wasmi_fuel.saturating_sub(self.engine_consumed);
+		if !wasmi_fuel.is_zero() {
+			self.engine_consumed += wasmi_fuel;
```

Comment:
well, it is safe because 
```
		wf = wf_old - ec_old;
		ec = ec_old + wf = ec_old + wf_old - ec_old = wf_old;
```

but yeah, this operation is excessive here. Re-wrote it in a (hopefully) more clear way.

Reply:
> I think charge_fuel is still the correct name because sync suggest that it may also sync stuff to the engine. But it is strictly one way information flow (removing stuff from the weight meter).

I'm not opposing the either name, just curious though, why do you think "sync" has to be two-dimensional? It could be one way sync, which seems legit enough to me as well.